3-Solving-Problems-By-Searching/c_breadthFirstSearch.js

```diff
@@ -1,54 +1,32 @@
-
-$(document).ready(function(){
-	var w = 600, h = 350;
-	var visGraph = null;
-	var visQueue = null;
-	var agent = null;
-	var initial = 0;
-	var end = 16;
-	var canvas = null;
-	var queueCanvas = null;
+$(document).ready(function() {
+  var w = 600,
+    h = 350;
 	var DELAY = 2000;
-	var updateFunction = null;
-	var intervalFunction = null;
-	var nextNodeColor = 'hsl(108, 96%, 50%)';
-	function init(){
-		canvas = document.getElementById('breadthFirstSearchCanvas');
-		queueCanvas = document.getElementById('fifoQueueCanvas');
-		graph = new makeDefaultGraph();
-		agent = new nodeExpansionAgent(graph.adjMatrix,initial);
-		visGraph = new drawGraph(canvas,h,w,agent,graph.nodes,graph.adjMatrix);
-		visQueue = new drawQueue(queueCanvas,h,w,agent,graph.nodes);
-		visGraph.init();
-		visQueue.init();
-		visGraph.nodeGroups[initial].children[0].fill = nextNodeColor;
-		visQueue.rectangles[0].fill = nextNodeColor;
-		visGraph.two.update();
-		visQueue.two.update();
-		updateFunction = function(){
-			frontier = agent.frontier;
-			if(frontier.length == 0){
+  function init() {
+    var graph = new DefaultGraph();
+    var graphProblem = new GraphProblem(graph.nodes,graph.edges,'A','A');
+    var graphAgent = new GraphAgent(graphProblem);
+    var options = new DefaultOptions();
+    options.nodes.next.fill = 'hsla(126, 100%, 69%, 1)';
+    var graphDrawAgent = new GraphDrawAgent(graphProblem,'breadthFirstSearchCanvas',options,h,w);
+    var queueDrawAgent = new QueueDrawAgent('fifoQueueCanvas',h,w,graphProblem,options);
+		var updateFunction = function() {
+			if(graphProblem.frontier.length > 0){
```

Comment:
Would you check what happened here? The indentation looks wrong on github. Is your text editor using tabs or spaces?